oximeter/collector/src/agent.rs

```diff
@@ -341,11 +350,13 @@ async fn collection_loop(
                             log,
                             "collection task received explicit request to collect"
                         );
+                        details.start_collection();
```

Comment:
This doesn't seem right to me for a couple reasons:

1. We aren't actually starting a collection here; we're attempting to enqueue a request to start a collection, which is a difference that seems important when debugging
2. If the timer fires immediately before or after this, we could call `start_collection()` twice before either `on_*()` completes, since we have two independent queues

I'm not entirely sure how to suggest reworking this (although this seems related to my comment above about making `start_collection()` return a handle, since that might at least let us avoid interleaving). Do we care about distinguishing how much of the time spent collecting was spent in the internal queue vs the actual collection request?

Reply:
> Do we care about distinguishing how much of the time spent collecting was spent in the internal queue vs the actual collection request?

I'm not sure, but it seems like it's not super tricky to get that and it's strictly more information. I'll rework this to handle the other comment, and figure out how to include this bit too.